gui/layers/_image_layer.py

```diff
@@ -270,3 +272,45 @@ def interpolation_functions(self):
 
     def _clim_range_default(self):
         return [np.min(self.image), np.max(self.image)]
+
+    def get_value(self, position, indices):
+        """Returns coordinates, values, and a string
+        for a given mouse position and set of indices.
+
+        Parameters
+        ----------
+        position : sequence of two int
+            Position of mouse cursor in canvas.
+        indices : sequence of int or slice
+            Indices that make up the slice.
+
+        Returns
+        ----------
+        coord : sequence of int
+            Position of mouse cursor in data.
+        value : int or float or sequence of int or float
+            Value of the data at the coord.
+        msg : string
+            String containing a message that can be used as
+            a status update.
+        """
+        transform = self._node.canvas.scene.node_transform(self._node)
+        pos = transform.map(position)
+        pos = [clip(pos[1],0,self.shape[0]-1), clip(pos[0],0,self.shape[1]-1)]
+        coord = copy(indices)
+        coord[0] = int(pos[0])
+        coord[1] = int(pos[1])
+        value = self._slice_image(coord)
+        msg = f'{coord}'
+        msg = msg + ', %s' % self.name + ', value '
+        if isinstance(value, ndarray):
+            if isinstance(value[0], integer):
+                msg = msg + '(%d, %d, %d)' % (value[0], value[1], value[2])
+            else:
+                msg = msg + '(%.3f, %.3f, %.3f)' % (value[0], value[1], value[2])
```

Comment:
Here (and elsewhere), we don't want `.3f` but rather `0.3`:

```python
In [7]: value = (3.141592, 2.718, 6.6666666667)
In [8]: '(' + str.join(', ', [f'{value[i]:0.3}' for i in range(3)]) + ')'
Out[8]: '(3.14, 2.72, 6.67)'
```

It will correctly round to three significant digits, regardless of the magnitude.

As to syntax, my preference is to always use f-strings, even if they are a bit clunky, because they read better, but otherwise, here you should at least do:

```suggestion
                msg += '(%0.3, %0.3, %0.3)' % value[:3]
```

Reply:
Switched to `0.3` and f-strings. Good catch for times when your image range is very very small, it is now doing the right thing